cdap-ui/app/directives/search/search-ctrl.js

```diff
@@ -15,21 +15,120 @@
  */
 
 angular.module(PKG.name + '.commons')
-  .controller('MySearchCtrl', function($state) {
-    this.searchTxt = '';
-    this.displaySearchBox = false;
-
-    this.onSearch = function(event) {
-      if (event.keyCode === 13) {
-        $state.go('search.objectswithtags', {tag: this.searchTxt});
-        this.searchTxt = '';
-        this.displaySearchBox = false;
-      }
+  .controller('MySearchCtrl', function($stateParams, myTagsApi, $timeout, $scope, $document) {
+    var vm = this;
+
+    vm.showSearchBox = false;
+    vm.searchTerm = '';
+    vm.searchResults = [];
+    vm.searchEntered = false;
+    vm.loading = false;
+
+    vm.showSearch = function () {
+      vm.showSearchBox = true;
+      $timeout(function () {
+        angular.element(document.getElementById('global-search'))[0].focus();
+      });
+    };
+
+    vm.onBlur = function () {
+      vm.showSearchBox = false;
+      vm.searchEntered = false;
+      vm.loading = false;
+      vm.searchTerm = '';
+      vm.searchResults = [];
     };
-    this.escapeSearch = function () {
-      if (this.searchTxt === '') {
-        this.displaySearchBox = false;
+
+    vm.search = function (event) {
+      event.stopPropagation();
+      if (event.keyCode === 27) {
+        vm.onBlur();
       }
+
+      if (!vm.searchTerm.length || event.keyCode !== 13) { return; }
+
+      vm.searchEntered = true;
+      vm.loading = true;
+      var params = {
+        namespaceId: $stateParams.namespace || $stateParams.nsadmin,
+        query: vm.searchTerm
+      };
+      myTagsApi.searchTags(params)
+        .$promise
+        .then(function (res) {
+          vm.loading = false;
+          var parsedSearch = [];
+          angular.forEach(res, function (entity) {
+            parsedSearch.push(parseEntity(entity.entityId));
+          });
+
+          vm.searchResults = parsedSearch;
+        });
+
     };
 
+    $scope.$on('$stateChangeSuccess', vm.onBlur);
```

Comment:
on state change wouldn't the controller automatically destroy everything? do we need to do this?